platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
I was actually thinking an overload of ::poll. Modern C++ style is to add overloads in the main namespace for functions to take your types (eg `std::swap(mbed::foo &, mbed::foo &)`).

People do want that definite poll functionality, but I don't think they're ever going to get it from sockets, as long as we are planning to support off-chip modules, which we definitely are. It would be a trap. Anyone writing code to rely on poll being accurate would be writing fragile unportable code. (Albeit many are managing that with sigio too...)

Given that people haven't been able to figure out the totally standard behaviour of SIGIO, I don't know how we would make them understand non-standard false positives from poll().

Even if we could make poll reliable, I fear it might be incredibly inefficient on some modules. Making a socket call to find out if you should make a socket call doesn't seem brilliant.

I agree that _poll_change shouldn't be a member - (I thought I'd already brought up that thought)

Will ponder your poll.